test/framework/src/main/java/org/elasticsearch/test/BuildUtils.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test;
+
+import org.elasticsearch.Build;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.RecordComponent;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.elasticsearch.test.ESTestCase.randomAlphaOfLength;
+import static org.elasticsearch.test.ESTestCase.randomFrom;
+import static org.elasticsearch.test.ESTestCase.randomInt;
+import static org.elasticsearch.test.ESTestCase.randomValueOtherThan;
+
+public class BuildUtils {
+    private static final MethodHandle buildCtor;
+    static {
+        try {
+            buildCtor = MethodHandles.publicLookup().unreflectConstructor(Build.class.getConstructors()[0]);
+        } catch (IllegalAccessException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    // creates a new Build instance with the same members, except those in extraArgs are substituted
+    public static Build newBuild(Build existing, Map<String, Object> extraArgs) {
```

Comment:
Rather than using reflection matching string names (in a way that refactorings wont catch), could we have an actual Builder in Build (or here) such that lambdas can be used to set the field names in code?

Reply:
I think a builder is a much more natural way to express something like this, and has been used in many other places in the ES codebase

Reply:
I agree builders are used in many other parts of ES, but I don't think that is a reason we should be forced to use that methodology in every place. For records, I see this as a much cleaner and more automatic way to create and mutate instances for tests.

Reply:
I think that using reflection & strings is a somewhat convoluted way of doing it, where a builder class would be a lot clearer.

Maybe we need another opinion from someone else on the team here?

Reply:
@ChrisHegarty Do you have any thoughts here?